stripe/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -176,10 +208,10 @@ internal class PaymentSheetViewModel internal constructor(
         val paymentSelection = selection.value
 
         if (paymentSelection is PaymentSelection.GooglePay) {
-            paymentIntent.value?.let { paymentIntent ->
+            stripeIntent.value?.let { stripeIntent ->
```

Comment:
Can you do `(stripeIntent.value as? PaymentIntent)?.let {}` to avoid the cast inside the lambda?